src/Common/Background/BasePeriodicBackgroundService.cs

```diff
@@ -18,7 +18,7 @@ public abstract class BasePeriodicBackgroundService : BackgroundService
 
     private Task? _executingTask;
 
-    private readonly TimeProvider _timeProvider;
+    protected readonly TimeProvider TimeProvider;
```

Comment:
Turn into a property for consistency?

```suggestion
    protected TimeProvider TimeProvider { get; }
```